### PR TITLE
doc: Add manual test for the cluster restore issue

### PR DESCRIPTION
**Pull request by shuo-wu**
longhorn/longhorn#2228

### CODE REVIEW

**Comment by khushboo-rancher** [comment]
1. What will be the implications of nodes added/deleted after the snapshot was taken? If replicas are evicted and node is deleted and later replicas/volumes got moved to new node, will the volume lose for the data?

2. If the cluster config is changed like node OS etc?

3. User has upgraded the Longhorn system after the snapshot and new crds/settings got added. What will happen if user restore to that snapshot where the Longhorn didn't have those crds/settings?

File: docs/content/manual/pre-release/cluster-restore/backup-block-deletion.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+4. Do the followings before the restore:
+    1. Delete volume A.
+    2. Write more data to volume B and create more backups.
+    3. Remove all current replicas one by one for volume C. Then all replicas of volume C are new replicas.
+    4. Remove some replicas for volume D. Do snapshot creation, deletion, and revert.
+    5. Sacle down the workload. Upgrade volume E from the default image to another engine image. And do expansion.
+    6. Create and attach volume F via UI. Write some data and do some snapshot operations. (Validate 1 case: Users need to manuall recover the volume if it's created after the cluster snapshot)
+5. Restore the cluster. 
+6. Check the followings according to the doc:
+    1. Volume A is back. But there is no data in it. And users can re-delete it.
```
Comment (jenting) [inline docs/content/manual/pre-release/cluster-restore/backup-block-deletion.md:25]: I do not understand why no data in volume A after the cluster snapshot restore.

File: docs/content/manual/pre-release/cluster-restore/backup-block-deletion.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    3. Deploy a StatefulSet with volume C. Write some data and do some snapshot operations. (Validate 1 case: Users need to manually recover the volume if all existing replicas are replaced by new replicas)
+    4. Deploy a StatefulSet with volume D. Write some data and do some snapshot operations. (Validate 2 cases: <1> volume can be recovered automatically if some replicas are removed and some new replicas are replenished; <2> snapshot info will be resynced;)
+    5. Deploy a Deployment with volume E. Write some data and do some snapshot operations. (Validate 4 cases: <1> engine upgrade; <2> offline expansion)
+3. Create a cluster snapshot via Longhorn.
+4. Do the followings before the restore:
+    1. Delete volume A.
+    2. Write more data to volume B and create more backups.
+    3. Remove all current replicas one by one for volume C. Then all replicas of volume C are new replicas.
+    4. Remove some replicas for volume D. Do snapshot creation, deletion, and revert.
+    5. Sacle down the workload. Upgrade volume E from the default image to another engine image. And do expansion.
```
Comment (c3y1huang) [inline docs/content/manual/pre-release/cluster-restore/backup-block-deletion.md:21]: Scale

File: docs/content/manual/pre-release/cluster-restore/backup-block-deletion.md (hunk 1 above)
Comment (shuo-wu) [reply docs/content/manual/pre-release/cluster-restore/backup-block-deletion.md:25]: Volume A is actually removed in step4-1 and all the data is cleaned up. The cluster restore just recover the CR objects of this volume only.

**Comment by shuo-wu** [comment]
@khushboo-rancher Thank you for pointing out the missed cases! I have updated both the doc and the manual tests for the node creation/deletion case and the system upgrade case. As for the OS change, I don't think we need to consider it since we assume users are reverting/restoring a snapshot in an existing cluster, rather than a new cluster with new nodes and different OS.

**Comment by c3y1huang** [comment]
@shuo-wu could you fix the typo in file name `sysmtem-upgrade-with-restore.md `?

**Comment by shuo-wu** [comment]
> @shuo-wu could you fix the typo in file name `sysmtem-upgrade-with-restore.md `?

Fixed. Thank you!

File: docs/content/manual/pre-release/cluster-restore/node-creation-deletion-with-restore.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+title: Node creation and deletion with restore
+---
+
+Notice that the behaviors will be different if the cluster node roles are different. e.g., A cluster contains 1 dedicated master node + 3 worker node is different from a cluster contains 3 nodes which are both master and worker.
+This test may need to be validated for both kind of cluster.
+
+1. Deploy a 3-worker-node cluster then install Longhorn system.
+2. Deploy some workloads using Longhorn volumes then write some data.
+3. Create a cluster snapshot via Rancher.
+4. Add a new worker node for this cluster. Deploy workloads on this node.
```
Comment (khushboo-rancher) [inline docs/content/manual/pre-release/cluster-restore/node-creation-deletion-with-restore.md:11]: Do you mean, Deploy workloads attached with new volumes?

File: docs/content/manual/pre-release/cluster-restore/node-creation-deletion-with-restore.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+5. Restore the cluster. 
+6. Follow the doc after the restore. Verify:
```
Comment (khushboo-rancher) [inline docs/content/manual/pre-release/cluster-restore/node-creation-deletion-with-restore.md:13]: Do we want to add doc link here?

File: docs/content/manual/pre-release/cluster-restore/system-upgrade-with-restore.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+title: Longhorn system upgrade with restore
+---
+
+Notice that the behaviors will be different if the cluster node roles are different. e.g., A cluster contains 1 dedicated master node + 3 worker node is different from a cluster contains 3 nodes which are both master and worker.
+This test may need to be validated for both kind of cluster.
+
+1. Install Longhorn system. Deploy one more compatible engine image.
+2. Deploy some workloads using Longhorn volumes then write some data.
+3. Create a cluster snapshot via Rancher.
+4. Upgrade the Longhorn system to a newer version. Then modify the settings or node configs (especially the configs introduced in the new version).
```
Comment (khushboo-rancher) [inline docs/content/manual/pre-release/cluster-restore/system-upgrade-with-restore.md:11]: Should be validate the new volumes created with the upgraded engine at this step should disappear after the restore and doesn't create any problem?

**Review by khushboo-rancher — COMMENTED** [review]
LGTM in general.

File: docs/content/manual/pre-release/cluster-restore/node-creation-deletion-with-restore.md (hunk 3 above)
Comment (shuo-wu) [reply docs/content/manual/pre-release/cluster-restore/node-creation-deletion-with-restore.md:11]: Yes

File: docs/content/manual/pre-release/cluster-restore/node-creation-deletion-with-restore.md (hunk 4 above)
Comment (shuo-wu) [reply docs/content/manual/pre-release/cluster-restore/node-creation-deletion-with-restore.md:13]: Yes. But right now there is no URL for the doc. Maybe we need to wait for the new version doc released first?

File: docs/content/manual/pre-release/cluster-restore/system-upgrade-with-restore.md (hunk 5 above)
Comment (shuo-wu) [reply docs/content/manual/pre-release/cluster-restore/system-upgrade-with-restore.md:11]: I was thinking this step may be duplicate. If it's better to have, I will add it then.

**Review by innobead — APPROVED** [review]
LGTM

**Pull request merged**